agent/engine/serviceconnect/manager_linux.go

```diff
@@ -372,17 +370,30 @@ func (m *manager) AugmentInstanceContainer(task *apitask.Task, container *apicon
 	return nil
 }
 
-// LoadImage helps load the AppNetAgent container image for the agent
+// LoadImage helps load the AppNetAgent container image for the agent latest supported
+// AppNet interface version by looking for the AppNet agent tar name from supported list
+// of AppNet versions from highest to lowest version when loading AppNet image
 func (agent *manager) LoadImage(ctx context.Context, _ *config.Config, dockerClient dockerapi.DockerClient) (*types.ImageInspect, error) {
-	logger.Debug("Loading appnet agent container tarball:", logger.Fields{
-		field.Image: agent.AgentContainerTarballPath,
-	})
-	if err := loader.LoadFromFile(ctx, agent.AgentContainerTarballPath, dockerClient); err != nil {
-		return nil, err
+	var err error
+	for _, supportedAppnetInterfaceVersion := range getSupportedAppnetInterfaceVersions() {
+		agentContainerTarballPath := fmt.Sprintf(defaultAgentContainerTarballPathFormat, supportedAppnetInterfaceVersion)
+		if _, err = os.Stat(agentContainerTarballPath); err == nil {
```

Comment:
Maybe also check for non-nil error here to be more consistent with load error below?
```
if _, err = os.Stat(agentContainerTarballPath); err != nil {
 // log a warning message
 continue
}
if loadErr...
```

Reply:
Will add a check and a warning log here, thanks!